### PR TITLE
FIX 37847 New feature is hidden when created from attribute table

### PR DESCRIPTION
**Pull request by suricactus**
Now when a new feature is added in the attribute table when in form view, the new feature is automatically selected and the feature list on the left side is ensured to show it (not necessarily on top). In response of #37847 .

Peek 2020-09-09 09-45

### CODE REVIEW

File: src/gui/attributetable/qgsfeaturelistview.cpp [hunk 1]
```diff
@@ -189,18 +189,27 @@ void QgsFeatureListView::selectAll()
...
   const auto constFids = fids;
   for ( QgsFeatureId fid : constFids )
   {
-    selection.append( QItemSelectionRange( mModel->mapToMaster( mModel->fidToIdx( fid ) ) ) );
+    QModelIndex modelIdx = mModel->fidToIdx( fid );
+
+    if ( ! firstModelIdx.isValid() )
+      firstModelIdx = modelIdx;
+
+    selection.append( QItemSelectionRange( mModel->mapToMaster( firstModelIdx ) ) );
```
Comment (signedav) [inline src/gui/attributetable/qgsfeaturelistview.cpp:202]: Not sure, but shouldn't it take `modelIdx` still?

Comment (suricactus) [reply src/gui/attributetable/qgsfeaturelistview.cpp:202]: thanks, true that

File: src/app/qgsfeatureaction.h [hunk 2]
```diff
@@ -61,6 +61,12 @@ class APP_EXPORT QgsFeatureAction : public QAction
      */
     void setForceSuppressFormPopup( bool force );
 
+    /**
+     * Returns the current feature. It might be nullptr feature, so it's safe to use this method only
+     * right after`addFeature()` returned true.
+     */
+    QgsFeature *feature() const { return mFeature; }
```
Comment (nyalldawson) [inline src/app/qgsfeatureaction.h:68]: Can we return a invalid QgsFeature here instead of mFeature is null, and avoid the raw pointer please?

Comment (suricactus) [reply src/app/qgsfeatureaction.h:68]: Done!

**Review by signedav — APPROVED** [review]

**Pull request merged**